sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCOptions.scala

```diff
@@ -35,7 +50,12 @@ class JDBCOptions(
   val table = parameters("dbtable")
 
   // ------------------------------------------------------------
-  // Optional parameter list
+  // Optional parameters
+  // ------------------------------------------------------------
+  val driver = parameters.get(JDBCOptions.JDBC_DRIVER_CLASS)
```

Comment:
I do favour to keep the validation logics and options within `JDBCOptions` if you don't strongly feel about moving it into `JdbcUtils.createConnectionFactory`.